index.js

```diff
@@ -1,84 +1,173 @@
+            daysOfTheWeek: urls[name].daysOfTheWeek
+        }
+        const mentionsWhenDown = urls[name].mentionsWhenDown;
+        createLoop(name, url, interval, dateBounds, mentionsWhenDown);
     }
 });
 
 /* Create a loop that's executed every x milliseconds */
-function createloop(name, url, interval) {
-    var requestLoop = setInterval(function () {
-        request({
+function createLoop(name, url, interval, dateBounds, mentionsWhenDown) {
+    setInterval(function () {
+        const requestDate = new Date();
+        if (!isDateWithinBounds(requestDate, dateBounds)) {
+            urlContainer[name] = undefined;
+            return;
+        }
+
+        request.get({
             url: url,
-            method: "GET",
-            timeout: 10000,
-            followRedirect: true,
-            maxRedirects: 10
-        }, function (error, response, body) {
-            if (!error && response.statusCode == 200) {
-                if (typeof notReachable[name] != "undefined") {
-                  if (notReachable[name] == "notagain") {
-                    delete notReachable[name];
-                    var message = name + " " + emoji.get('beers') + " " + getFormattedDate();
-                    send(message);
-                    console.log(message);
-                  }
-                }
+            timeout: 2000,
+            followRedirect: true
+        }, (error, response, body) => {
+            const responseDate = new Date();
+
+            if (error !== null) {
+                handleLocalError(name, responseDate);
+            } else if (response.statusCode == 200) {
+                handleSuccessStatus(name, url, responseDate);
             } else {
-                if (typeof notReachable[name] == "undefined") { // first time
-                    notReachable[name] = "true";
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log("first time :" + message);
-
-                } else if (notReachable[name] == "true") { // secondtime
-                    notReachable[name] = "notagain"; // we don't want second notification
-                    var message = name + " " + emoji.get('hankey') + " " + getFormattedDate();
-                    console.log(message);
-                    send(message);
-                }
+                handleDownStatus(name, url, responseDate, mentionsWhenDown);
             }
         });
     }, interval);
 }
 
+function handleSuccessStatus(urlName, urlValue, responseDate) {
+    const message = `:globe_with_meridians: ${urlName} (${urlValue}) is up and running!`;
+
+    if (urlContainer[urlName] === undefined) {
+        send("First check of this session... Endpoint is running! :beer:");
+    } else {
+        if (urlErrorContainer[urlName] !== undefined) {
+            restoreLocalErrorAndSend(responseDate, urlName);
+        }
+
+        if (urlContainer[urlName] !== UrlStatus.Up) {
```

Comment:
Didn't get this part, if the status is not up we are sending a message saying that it is up and running?

Reply:
We haven't set the status in the container at this point. The `handleSuccessStatus` assumes that the ping was successful and checks the previous status (which is what's in the container). After that it sets the status in the container to Up